mod_reforged/hooks/skills/effects/bleeding_effect.nut

```diff
@@ -0,0 +1,44 @@
+::Reforged.HooksMod.hook("scripts/skills/effects/bleeding_effect", function(q) {
+	q.m.Attacker <- null;
+
+	q.onAdded = @(__original) function()
+	{
+		__original();
+		if (!this.isGarbage())
+		{
+			this.m.Attacker = this.getContainer().getActor().getLastAttacker();
+		}
+	}
+
+	q.applyDamage = @() function()
+	{
+		if (this.m.LastRoundApplied != ::Time.getRound())
+		{
+			this.m.LastRoundApplied = ::Time.getRound();
+			local actor = this.getContainer().getActor();
+			this.spawnIcon("status_effect_01", actor.getTile());
+			local hitInfo = clone ::Const.Tactical.HitInfo;
+			hitInfo.DamageRegular = this.m.Damage * (actor.getSkills().hasSkill("effects.hyena_potion") ? 0.5 : 1.0);
+			hitInfo.DamageDirect = 1.0;
+			hitInfo.BodyPart = ::Const.BodyPart.Body;
+			hitInfo.BodyDamageMult = 1.0;
+			hitInfo.FatalityChanceMult = 0.0;
+			actor.onDamageReceived(this.getAttacker(), this, hitInfo);
+
+			if (--this.m.TurnsLeft <= 0)
+			{
+				this.removeSelf();
+			}
+		}
+	}
+
+	q.getAttacker <- function()
+	{
+		if (!::MSU.isNull(this.m.Attacker) && this.m.Attacker.isAlive() && this.m.Attacker.isPlacedOnMap())
```

Comment:
Is `this.m.Attacker.isAlive()` important to prevent any crashes?

Right now:
If one of my brothers dies after applying a bleed their bleed wouldn't contribute towards my team gaining experience or potentially the loot of that enemy.

Reply:
I believe the `isAlive()` check is important because if this actor dies as as result of this damage, then it will trigger a chain of events such as `onActorKilled` and others for the killer. When I was working on Legends I saw that there were comments in the `bleeding_effect` code file saying that crashes were happening unless you add both `isAlive()` and `isPlacedOnMap()` checks.

> If one of my brothers dies after applying a bleed their bleed wouldn't contribute towards my team gaining experience or potentially the loot of that enemy.
This is a valid concern. I will try to think if we can solve this.